.github/workflows/publish-go-tester-task.yml

```diff
@@ -35,14 +35,19 @@ jobs:
     outputs:
       result: ${{ steps.determination.outputs.result }}
     steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
       - name: Determine if the rest of the workflow should run
         id: determination
         run: |
-          RELEASE_BRANCH_REGEX="refs/heads/[0-9]+.[0-9]+.x"
+          TAG="$(git tag --points-at=HEAD 2> /dev/null | head -n1)"
           # The `create` event trigger doesn't support `branches` filters, so it's necessary to use Bash instead.
           if [[ \
-            "${{ github.event_name }}" != "create" || \
-            "${{ github.ref }}" =~ $RELEASE_BRANCH_REGEX \
+            "${{ github.event_name }}" != "create" && \
+            "$TAG" == "" \
```

Comment:
This logic makes it so the other jobs in the workflow will never run on a `create` event trigger. The reason that event trigger and this logic was added is explained here: https://github.com/arduino/tooling-project-assets/pull/148

A full run of the workflow must occur on any of the following events:

- Creation of a release branch (`create` event)
- Commit is pushed that changes relevant files (`push` event)
- PR creation/sync that changes relevant files (`pull_request` event)
- Manual triggers (`workflow_dispatch`/`repository_dispatch` event)

For the sake of efficiency, we specifically want to avoid the full run of the workflow on the following events:

- Creation of a non-release branch
- Commit/PR that doesn't change relevant files

So this PR must find a way to preserve all the above behavior while also achieving the following:

Avoid full run of workflow on `push` event resulting from a tag push.

Reply:
I modified the check. Now the workflow runs if a release branch is created (https://github.com/MatteoPologruto/arduino-cli/actions/runs/3593113796) but the run is skipped if a tag is pushed (https://github.com/MatteoPologruto/arduino-cli/actions/runs/3592812114).